examples/transfer_media.py

```diff
@@ -19,143 +20,369 @@
+        action='store_true',
     )
 
     args = parser.parse_args()
 
-    try:
-        media_download_dir = args.temp_path / args.oid
-        media_download_dir.mkdir(parents=True)
+    msc_src = MediaServerClient(args.conf_src)
+    src_domain = msc_src.conf['SERVER_URL'].split('/')[2]
+    external_ref_prefix = f'nudgis:{src_domain}'
 
-        msc_src = MediaServerClient(args.conf_src)
+    msc_dest = MediaServerClient(args.conf_dest)
 
-        zip_path = backup_media(msc_src, args.oid, media_download_dir)
-        print(f"media {args.oid} downloaded to {zip_path}")
+    dest_videos = msc_dest.get_catalog(fmt='flat').get('videos', list())
+    dest_external_refs = dict()
+    for v in dest_videos:
+        dest_external_refs[v['external_ref']] = v['oid']
+
+    oids_src = list()
+
+    if args.oid_file and args.oid_file.is_file():
+        print(f'Reading oids from {args.oid_file}')
+        oids_src = args.oid_file.open().read().strip().split('\n')
+    if args.oid:
+        oids_src += args.oid
+
+    oid_src_count = len(oids_src)
+    if oid_src_count:
+        print(f'Found {oid_src_count} oids to transfer')
+    else:
+        sys.exit('No oids provided, exiting')
+
+    failed = list()
+    skipped = list()
+    done = list()
+
+    for index, oid_src in enumerate(oids_src):
+        print(f'Processing {index + 1}/{oid_src_count}')
+
+        external_ref = f'{external_ref_prefix}:{oid_src}'
+        if oid_dst := dest_external_refs.get(external_ref):
+            print(
+                f'Media {external_ref} already uploaded as {oid_dst}, skipping source media {oid_src}'
+            )
+            skipped.append(oid_src)
+            continue
+
+        media_download_dir = args.temp_path / oid_src
+        try:
+            zip_path = backup_media(msc_src, oid_src, media_download_dir)
+        except Exception as e:
+            error = f"Failed to backup media {oid_src}: {e}, ignoring for now"
+            print(error)
+            failed.append(error)
+            continue
 
         def print_progress(progress):
-            print(f"Uploading: {progress * 100:.1f}%", end="\r")
+            print(f'Uploading: {progress * 100:.1f}%', end='\r')
 
-        msc_dest = MediaServerClient(args.conf_dest)
-        print("Starting upload")
-        resp = msc_dest.add_media(file_path=zip_path, progress_callback=print_progress)
+        upload_args = {
+            'file_path': zip_path,
+            'progress_callback': print_progress,
+            'external_ref': external_ref,
+            'own_media': 'no',
+        }
 
-        if resp["success"]:
-            print(f"File {zip_path} upload finished, object id is {resp['oid']}")
+        metadata = extract_metadata_from_zip(zip_path)
+        src_path = metadata['path']
+
+        is_personal_channel = False
+        if args.migrate_personal_channels and metadata.get('speaker'):
+            if args.personal_channels_root in src_path:
+                is_personal_channel = True
```

Comment:
It may be a sub channel of the personal channel, so this variable should be named `in_personal_channel`.